xarray/tests/test_dataset.py

```diff
@@ -2719,6 +2719,12 @@ def test_where_drop_empty(self):
         expected = DataArray(np.zeros((0, 10)), dims=['nCells', 'nVertLevels'])
         self.assertDatasetIdentical(expected, actual)
 
+    def test_where_drop_no_indexes(self):
+        ds = Dataset({'foo': ('x', [0.0, 1.0])})
+        expected = Dataset({'foo': ('x', [1.0])})
+        actual = ds.where(ds == 1, drop=True)
+        self.assertDatasetIdentical(expected, actual)
```

Comment:
+1, might be worth a large search/replace PR in the full test suite. Right now I'm pretty sure `assertDatasetIdentical` is just an alias to `assert_identical`.